x/eibc/keeper/invariants.go

```diff
@@ -73,3 +74,35 @@ func UnderlyingPacketExistInvariant(k Keeper) sdk.Invariant {
 		return sdk.FormatInvariant(types.ModuleName, "underlying-packet-exist", msg), broken
 	}
 }
+
+func CoinsInvariant(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		var (
+			broken bool
+			msg    string
+		)
+		allDemandOrders, err := k.ListAllDemandOrders(ctx)
+		if err != nil {
+			msg += fmt.Sprintf("list all demand orders failed: %v\n", err)
+			broken = true
+		}
+		for _, do := range allDemandOrders {
+			for _, coins := range []sdk.Coins{do.Price, do.Fee} {
+				if len(coins) == 0 {
+					// This is OK, since coins will erase the zero coin and zero price/fee is allowed
+					continue
+				}
+				if len(coins) > 1 {
+					msg += fmt.Sprintf("multiple coins: %s\n", coins)
+					broken = true
+					continue
+				}
+				if coins[0].IsNegative() {
+					msg += fmt.Sprintf("negative coins: %s\n", coins)
+					broken = true
+				}
+			}
+		}
+		return sdk.FormatInvariant(types.ModuleName, "coins", msg), broken
+	}
+}
```

Comment:
@zale144 please review

Reply:
where's the "Num pending + num finalized = num total" invariant?